SUSE/scf
# Alternate handling of vcap hard and soft nproc limits

Alternate fix for #2100 based on changes to fissile instead of SCF itself.

See https://github.com/SUSE/scf/pull/2155 for the previous PR we are reverting/replacing.
Fissile PR is https://github.com/cloudfoundry-incubator/fissile/pull/483

Ticket https://trello.com/c/XvU6E5Vg/955-allow-bumping-up-the-nproc-limit-for-vcap-user

## Test plan

- Deploy SCF without specifying `kube.limits.nproc.*` and:
  - Get a shell session to any container and:
    - `su vcap`
    - `ulimit -u` should return 4096 (the default value).
- Perform an upgrade setting only `kube.limits.nproc.hard`:
  - The containers should error.
    Grab the logs of the failed container (secret-generation) with `kubectl logs -n cf secret-generation-* -p`.
    The error should say that `soft` must be set when `hard` is set.
- Repeat the previous point, but setting only `kube.limits.nproc.soft`.
  A similar error should be raised.
- Perform an upgrade setting both `soft` and `hard`, but set `soft` to a value higher than `hard`.
  It should fail with "`soft` (val1) cannot be larger than `hard` (val2)".
- Perform an upgrade setting both `hard` and `soft` to 8192, and:
  - Get a shell session to any container and:
    - `su vcap`
    - `ulimit -u` should return 8192 (the provided value).
    - `exit` (the su shell)
    - `cat /etc/security/limits.conf | grep nproc`.
      Check that both the hard and soft limits were set accordingly.
